test_util/integration_test.py

```diff
@@ -126,8 +126,8 @@ def _wait_for_slaves_to_join(self):
             return False
         data = r.json()
         num_slaves = len([x['hostname'] for x in data['slaves']])
-        # For single node setup there is only one slave node:
-        min_slaves = min(len(self.slaves), 2)
+        # There is one slave on every master, as well as slaves on agents.
+        min_slaves = len(self.slaves) + len(self.masters)
```

Comment:
The single node master + agent case we've gotten rid of, so there isn't the odd extra case needed. Might be worthwhile to globally fail early if not give at least two slaves, but that's a different condition